### PR TITLE
Plot.image rotate option

### LINKED ISSUE #1083: image should have a `rotate` option just like dot
It would be really handy for me if `image` marks had the `rotate` option just like `dot` marks do.

My particular use case is making a diamond chart, which requires rotating the whole plot, which necessitates rotating any of the elements that appear within.

### PR DESCRIPTION
**Pull request by Fil**
Demo: https://observablehq.com/@observablehq/plot-image-rotate-1084

closes #1083

todo:
- [ ] unit test
- [ ] dx, dy— & anticipate what will happen if dx and dy become variable (as in #909)

### CODE REVIEW

File: src/marks/image.js [hunk 1]
```diff
@@ -104,6 +107,19 @@ export class Image extends Mark {
...
+            R || this.rotate
+              ? X && Y
+                ? (i) => `${X[i]}px ${Y[i]}px`
+                : X
+                ? (i) => `${X[i]}px ${cy}px`
+                : Y
+                ? (i) => `${cx}px ${Y[i]}px`
+                : `${cx}px ${cy}px`
+              : null
+          )
```
Comment (Fil) [inline src/marks/image.js:122]: unfortunate complexity… am I missing a better way?

Comment (mbostock) [reply src/marks/image.js:122]: Can you set use transform-origin="center" instead?

Comment (mbostock) [reply src/marks/image.js:122]: I tried it and it doesn’t worker. ☹️ So I guess this is the only option… unless you wrap the image in a G element or something, but I bet that’s worse.

**Comment by llimllib** [comment]
It looks great to me, thanks for the quick response! Though I have to say I don't understand the `transform-origin` bit at all

**Review by llimllib — APPROVED** [review]

**Review by mbostock — COMMENTED** [review]
I think we want to combine the x, y, and transform-origin attributes into a single transform attribute, similar to what we do for dot and vector. Also this could use the `template` helper I created for #1189.

**Comment by mbostock** [comment]
This looks useful.

**Comment by Fil** [comment]
superseded by #1426

**Pull request closed without merging**